src/FrameDecorator/Page.php

```diff
@@ -357,14 +357,33 @@ protected function _page_break_allowed(Frame $frame)
 
                 // Rule C
                 $block_parent = $frame->find_block_parent();
-                if (count($block_parent->get_line_boxes()) < $frame->get_style()->orphans) {
+                $block_line_boxes = $block_parent->get_line_boxes();
+                $frame_line_number = 0;
+                $i = 1;
+                $first_node_on_line = false;
```

Comment:
The variable is indeed unused.

Reply:
Perhaps was meant to be used at some point but never fully developed. I think removing the fist-line logic makes sense since it's not current used. This should be easy enough to re-implement if we need it later.